301-hdinsight-windows-storm-iothub/azuredeploy.json

```diff
@@ -0,0 +1,267 @@
+      }
+    },
+    "iotFeedbackMessagesMaxDeliveryCount": {
+      "type": "int",
+      "minValue": 1,
+      "maxValue": 100,
+      "defaultValue": 10,
+      "metadata": {
+        "description": "Maximum delivery count for feedback queue."
+      }
+    },
+    "iotFeedbackMessagesLockDurationAsIso8601": {
+      "type": "string",
+      "defaultValue": "PT60S",
+      "metadata": {
+        "description": "Lock duration in ISO_8601 format for service-bound feedback messages. The value can range from PT5S to PT300S (or any other format as per ISO_8601 standards for duration)."
+      }
+    },
+
+    "stormClusterName": {
+      "type": "string",
+      "defaultValue": "storm-cluster",
+      "metadata": {
+        "description": "The name for the storm cluster."
+      }
+    },
+    "stormClusterLoginUserName": {
+      "type": "string",
+      "defaultValue": "admin",
+      "metadata": {
+        "description": "These credentials can be used to submit jobs to the cluster and to log into cluster dashboards."
+      }
+    },
+    "stormClusterLoginPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The password must be at least 10 characters in length and must contain at least one digit, one non-alphanumeric character, and one upper or lower case letter."
+      }
+    },
+    "stormRDPUserName": {
+      "type": "string",
+      "defaultValue": "winadmin",
+      "metadata": {
+        "description": "These credentials can be used to submit jobs to the cluster and to log into cluster dashboards."
+      }
+    },
+    "stormRDPPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The password must be at least 10 characters in length and must contain at least one digit, one non-alphanumeric character, and one upper or lower case letter."
+      }
+    }
+  },
+  "variables": {
+    "location": "[resourceGroup().location]",
+    "vNet": {
+      "name": "[concat(parameters('stormClusterName'),'-vnet')]",
+      "addressSpacePrefix": "10.0.0.0/16",
+      "subnetName": "subnet1",
+      "subnetPrefix": "10.0.0.0/24",
+      "id": "[resourceId('Microsoft.Network/virtualNetworks', concat(parameters('stormClusterName'),'-vnet'))]",
+      "subnet": "[concat(resourceId('Microsoft.Network/virtualNetworks', concat(parameters('stormClusterName'),'-vnet')), '/subnets/subnet1')]"
+    }
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[parameters('storageAccountName')]",
+      "apiVersion": "2015-06-15",
+      "location": "[variables('location')]",
+      "properties": {
+        "accountType": "[parameters('storageAccountType')]"
+      }
+    },
+
+    {
+      "name": "[parameters('stormClusterName')]",
+      "type": "Microsoft.HDInsight/clusters",
+      "location": "[variables('location')]",
+      "apiVersion": "2015-03-01-preview",
```

Comment:
@gatneil -
All examples in this repo with hdinsight uses 2015-03-01-preview.
When I try a different version I get:
`
{
  "error": {
    "code": "NoRegisteredProviderFound",
    "message": "No registered resource provider found for location 'westeurope' and API version '2016-01-01' for type 'clusters'. The supported api-versions are '2015-03-01-preview'. The supported locations are 'eastasia, southeastasia, eastus, eastus2, westus, northcentralus, southcentralus, centralus, northeurope, westeurope, japaneast, japanwest, brazilsouth, australiaeast, australiasoutheast, centralindia'."
  }
}
`